control/plugin/execution.go

```diff
@@ -167,12 +183,27 @@ func (e *ExecutablePlugin) Kill() error {
 func (e *ExecutablePlugin) captureStderr() {
 	stdErrScanner := bufio.NewScanner(e.stderr)
 	go func() {
-		for stdErrScanner.Scan() {
-			execLogger.
-				WithField("io", "stderr").
-				WithField("plugin", path.Base(e.cmd.Path())).
-				Debug(stdErrScanner.Text())
+		for {
+			for stdErrScanner.Scan() {
+				execLogger.
+					WithField("plugin", path.Base(e.cmd.Path())).
+					WithField("io", "stderr").
+					Debug(stdErrScanner.Text())
+			}
 
+			if err := stdErrScanner.Err(); err != nil {
+				reader := bufio.NewReader(e.stderr)
+				log, err := reader.ReadString('\n')
```

Comment:
we need to handle the err from `ReadString`

```
log, err2 := reader.ReadString('\n')
				if err2 == io.EOF {
					break
				}
```
Without the above block when the plugin stops the go routine never exits.  You can see this by loading the modified mock2 plugin and starting the following task and then finally stop (ctrl-c) snapd.
```
---
  version: 1
  schedule: 
    type: "simple"
    interval: "1s"
  max-failures: 10
  workflow: 
    collect: 
      metrics: 
        /intel/mock/foo: {}
        /intel/mock/bar: {}        
      config: 
        /intel/mock: 
          name: "root"
          password: "secret"          
          long_stderr_log: true
```
![img](https://www.evernote.com/l/AA0HS90G18hPDqv9aiOE4hOPYhkenqzW__8B/image.png)